containers-toolkit/Public/AllToolsUtilities.psm1

```diff
@@ -60,73 +64,117 @@ function Install-ContainerTools {
-    $nerdctlTarFile = "nerdctl-${nerdctlVersion}-windows-amd64.tar.gz"
-
-    # Installation paths
-    $ContainerdPath = "$InstallPath\Containerd"
-    $BuildkitPath = "$InstallPath\Buildkit"
-    $NerdCTLPath = "$InstallPath\Nerdctl"
-
-
-    $files = @(
-        [PSCustomObject]@{
-            Feature      = "Containerd"
-            Uri          = "https://github.com/containerd/containerd/releases/download/v$containerdVersion/$containerdTarFile"
-            Version      = $containerdVersion
-            DownloadPath = "$DownloadPath\$($containerdTarFile)"
-            InstallPath  = $ContainerdPath
-            EnvPath      = "$ContainerdPath\bin"
-        }
-        [PSCustomObject]@{
-            Feature      = "BuildKit"
-            Uri          = "https://github.com/moby/buildkit/releases/download/v${BuildKitVersion}/$BuildKitTarFile"
-            Version      = $BuildKitVersion
-            DownloadPath = "$DownloadPath\$($BuildKitTarFile)"
-            InstallPath  = $BuildkitPath
-            EnvPath      = "$BuildkitPath\bin"
+    begin {
+        $toInstall = @("containerd", "buildkit", "nerdctl")
+        $toInstallString = $($toInstall -join ', ')
+
+        $WhatIfMessage = "$toInstallString will be installed"
+        if ($Force) {
+            <# Action when this condition is true #>
+            $WhatIfMessage = "$toInstallString will be automativally uninstalled (if they are already installed) and reinstalled"
         }
-        [PSCustomObject]@{
-            Feature      = "Nerdctl"
-            Uri          = "https://github.com/containerd/nerdctl/releases/download/v${nerdctlVersion}/$nerdctlTarFile"
-            Version      = $nerdctlVersion
-            DownloadPath = "$DownloadPath\$($nerdctlTarFile)"
-            InstallPath  = $NerdCTLPath
-            EnvPath      = $NerdCTLPath
+        if ($CleanUp) {
+            <# Action when this condition is true #>
+            $WhatIfMessage += " and downloaded files will be removed"
         }
-    )
+    }
 
-    # Download files
-    Get-InstallationFiles -Files $files
-
-    # Install tools
-    foreach ($params in $files) {
-        try {
-            # Uninstall if tool exists at specified location. Requires user consent
-            Uninstall-ContainerTool -Tool $params.Feature -Path $params.InstallPath
-
-            # Untar downloaded files to the specified installation path
-            $InstallParams = @{
-                Feature      = $params.Feature
-                InstallPath  = $params.InstallPath
-                DownloadPath = $params.DownloadPath
-                EnvPath      = $params.EnvPath
+    process {
+        if ($PSCmdlet.ShouldProcess($InstallPath, $WhatIfMessage)) {
+            Write-Debug "Tools to install: $toInstallString"
+
+            # Global Variables needed for the script
+            $containerdTarFile = "containerd-${containerdVersion}-windows-amd64.tar.gz"
```

Comment:
nit: for all these versions, need to provide the architecture as a param since we are going to have ARM64s soon, hence, e.g. `containerd-${containerdVersion}-windows-${arch}.tar.gz`, where `arch = 'arm64|adm64'`.

Reply:
Created a new issue: #8